demos/sev-demo/README.md

```diff
@@ -0,0 +1,206 @@
+```
+# Path to the firmware.
+# If you want that qemu uses the default firmware leave this option empty
+firmware = "/<path-to-fw>/OVMF.fd"
+```
+
+## Rootfs
+ 
+Create the rootfs as you normally would for CCv0, but set `AA_KBC` to `offline_sev_kbc` so that the Attestation Agent is built with the correct key broker client. 
+
+```
+sudo make USE_DOCKER=yes DISTRO=ubuntu AGENT_INIT=yes DOCKER_RUNTIME=runc AA_KBC=offline_sev_kbc SKOPEO_UMOCI=yes rootfs
+```
+
+You will need to add the agent configuration file to the initrd. Again, this is a typical CCv0 requirement, but you will need to specify parameters for your KBC. Specifically be sure to add this parameter.
+```
+aa_kbc_params = "offline_sev_kbc::null"
+```
+
+The `offline_sev_kbc` uses the `efi_secret` kernel module to expose injected secrets to the filesystem. You'll need to add the kernel module to the initrd. You can do this by setting the `KERNEL_MODULES_DIR` when you build your rootfs or you can copy the kernel module directly into the rootfs that is created. Either way, your rootfs should look something like this. 
+```
+$ tree ubuntu_rootfs/lib/modules
+lib/modules
+└── 5.15.0-rc5+
+    ├── kernel
+    │   └── drivers
+    │       └── virt
+    │           └── coco
+    │               └── efi_secret
+    │                   └── efi_secret.ko
+    ├── modules.alias
+    ├── modules.alias.bin
+    ├── modules.builtin
+    ├── modules.builtin.alias.bin
+    ├── modules.builtin.bin
+    ├── modules.builtin.modinfo
+    ├── modules.dep
+    ├── modules.dep.bin
+    ├── modules.devname
+    ├── modules.order
+    ├── modules.softdep
+    ├── modules.symbols
+    └── modules.symbols.bin
+```
+
+Make sure that the module matches your kernel version. Speaking of which, `efi_secret` also requires some minor kernel patches. These patches and the module are not currently upstream. You can find the v6 patch [here](https://lore.kernel.org/lkml/20211129114251.3741721-1-dovmurik@linux.ibm.com/T/). Build a kernel with this patch and specify it in your Kata configuration file. This will also generate the module that should be specified above.
+
+Once you generate an initrd from the updated rootfs, update your Kata configuration file to point to the new initrd. You will also need to update the kernel field in the configuration file to point to your new kernel. For CCv0 you should have already added `agent.config_file=<agent-config-path>` to the `kernel_params` field. 
+
+## Kata Runtime
+
+Since SEV and SEV-ES use pre-attestation driven from outside of the guest, some changes are required for the Kata shim. These changes are not currently upstream. You will need to build your Kata shim with [this PR](https://github.com/kata-containers/kata-containers/pull/3025). This will work best with QEMU 6.2.0rc0 given that newer version of QEMU will expect to find the `kernel-hashes=on` flag, which is not currently set by the shim.
```

Comment:
@fitzthum should we add `kernel-hashes=on` to the shim PR?

Reply:
Yes.

Reply:
I'll add a link to the patch cover letter where this is explained in depth.